Cookbook/Technical-Documents/TheDebugWindow.md

```diff
@@ -0,0 +1,94 @@
+Integrate with the Debug Window
+==========================================
+
+Sometimes there is a complicated user journey, requiring multiple steps or data entry to proceed. Sometimes there is a prerequisite to reach a certain user flow that is awfully long for recurring manual QA and dev testing.
+
+You want to have an early way out to this prolonged, endured process of having to repeat yourselves just to get through them.
+
+## 😘 The Debug Window is here to help! 🎉
+
+## What is the Debug Window?
+
+The debug window can be activated in the following ways:
+
+* Swipe in from the right screen edge; or
+* Press Command + D if you are running the app in the Simulator.
+
+<img src="./Assets/thedebugwindow.png" width="200" />
+
+The Debug Window serves these functionalities:
+
+* Provide a list of contextual debug actions.
+* Quick access to useful debug information, like the current environment and the app version.
+
+## What are debug actions?
+
+Represented in code as `DebugAction`, these are debug shortcuts or side effects that are supported by your screen.
+
+All `DebugAction` have a title, which need not be localized. You may configure them to be either of these three forms:
+
+* Simple action — a simple one-off closure invocation;
+* Producer action — starting a `SignalProducer`, and showing an alert upon failure; and
+* Modal action — a factory closure which generates a `UIViewController`, which would be presented by the Debug Window.
+* Date action - A one-off closure that retrieves a selected (optional) date. You could also provide an initially selected date.
```

Comment:
Is this related to `DebugAction` ? 
It seems it relates to "debug window" alongside "gender picker" though.

Reply:
Sort of.

The date picker has been in there for a while (component showcase) is separate from any debug actions. I'm not sure if we need to keep it in there any longer to be honest.

A new `DebugAction` was added a few days ago that included this date action, which looks the same as the component showcase version, but can be used by screens to set certain date-related flags.

Reply:
I'm only seeing the type `DebugAction` and `DebugAction.Action` as a proof-of-concept, so I think we should not include information unrelated to these types.

Reply:
Sorry, I should clarify. `date` *is* a `DebugAction.Action` case. By "sort of", I meant that it looks identical to the component showcase picker (which has 0 function)

Reply:
OK, maybe we can move date-picker as another `DebugAction` with some special flag to place at the bottom? 🤔 

I still want to see it appearing in real code somehow, but for now, I think I understand the similar concept, so I will move on 👍

Reply:
This is already in code: #7228 (in Babylon repo)

Reply:
Oh I was looking at wrong branch... #7228 solves my issue 👍